qml/DomainGraphLayout.qml

```diff
@@ -0,0 +1,1462 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// This file is part of eProsima Fast DDS Monitor.
+//
+// eProsima Fast DDS Monitor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// eProsima Fast DDS Monitor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with eProsima Fast DDS Monitor. If not, see <https://www.gnu.org/licenses/>.
+
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import Theme 1.0
+
+Item
+{
+    id: domainGraphLayout
+
+    // Public properties
+    property var model: {}                          // domain view graph JSON model
+    property int entity_id                          // entity id associated to the domain id
+    property int domain_id                          // domain id
+    required property string component_id           // mandatory to be included when object created
+
+    // Public signals
+    signal update_tab_name(string new_name)         // Update tab name based on selected domain id
+
+    // Private properties
+    property var topic_locations_: {}               // topic information needed for connection representation
+    property var endpoint_topic_connections_: {}    // endpoint information needed for connection representation
+    property var topic_painted_: []                 // already painted topic connection references
+    property var endpoint_painted_: []              // already painted endpoint connection references
+    property int max_host_width_: 0                 // host entity box width management
+    property int max_user_width_: 0                 // user entity box width management
```

Comment:
Since `max_host_width_` is always equal to `max_user_width_ + 2*elements_spacing_`, and equal to `max_process_width_ + 4*elements_spacing_`, ..., there is no need to have several maximum widths, and a unique pass from endpoints to host would be enough to resize all elements.